staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -142,7 +142,7 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope *RequestSc
 		// that will break existing clients.
 		// Other cases where resource is not instantly deleted are: namespace deletion
 		// and pod graceful deletion.
-		if !wasDeleted && options.OrphanDependents != nil && !*options.OrphanDependents {
```

Comment:
Why did the items being checked suddenly change.  What was the rule that was failing?  The existing code doesn't seem obviously wrong.

Reply:
The field is marked deprecated, so static check wants usage to be removed. That is clearly wrong here, since this is the server code supporting the deprecated option and cannot stop referencing it.